scripts/covid_hosp_prod.R

```diff
@@ -97,226 +113,266 @@ rlang::list2(
-            mutate(time_value = time_value - 3)
-        }
-        nssp <- current_nssp_archive %>%
-          epix_as_of(min(forecast_date, current_nssp_archive$versions_end)) %>%
-          mutate(time_value = time_value)
-        attributes(train_data)$metadata$as_of <- as.Date(forecast_date_int)
-        train_data %>%
-          forecaster_fns[[forecasters]](ahead = aheads, extra_data = nssp) %>%
+  names = c("forecaster_fn_names", "forecast_date_chr"),
+  tar_target(
+    name = forecast_res,
+    command = {
+      if (as.Date(forecast_generation_date_int) < Sys.Date()) {
+        train_data <- nhsn_archive_data %>%
+          epix_as_of(min(as.Date(forecast_date_int), current_nssp_archive$versions_end)) %>%
+          add_season_info() %>%
           mutate(
-            forecaster = names(forecaster_fns[forecasters]),
-            geo_value = as.factor(geo_value)
+            geo_value = ifelse(geo_value == "usa", "us", geo_value),
+            time_value = time_value - 3
           )
-      },
-      pattern = cross(aheads, forecasters)
-    ),
-    tar_target(
-      name = ensemble_res,
-      command = {
-        forecast_res %>%
-          ensemble_linear_climate(
-            aheads,
-            other_weights = geo_forecasters_weights,
-            max_climate_ahead_weight = 0.6,
-            max_climate_quantile_weight = 0.6
-          ) %>%
-          filter(geo_value %nin% geo_exclusions) %>%
-          ungroup() %>%
-          sort_by_quantile()
-      },
-    ),
-    tar_target(
-      name = ensemble_mixture_res,
-      command = {
-        forecast_res %>%
-          ensemble_linear_climate(
-            aheads,
-            other_weights = geo_forecasters_weights,
-            max_climate_ahead_weight = 0.6,
-            max_climate_quantile_weight = 0.6
-          ) %>%
-          filter(geo_value %nin% geo_exclusions) %>%
-          ungroup() %>%
-          bind_rows(forecast_res %>%
-            filter(forecaster %in% c("windowed_seasonal_extra_sources")) %>%
-            filter(forecast_date < target_end_date)) %>% # don't use for neg aheads
-          group_by(geo_value, forecast_date, target_end_date, quantile) %>%
-          summarize(value = mean(value, na.rm = TRUE), .groups = "drop") %>%
-          sort_by_quantile()
-      },
-    ),
-    tar_target(
-      name = forecasts_and_ensembles,
-      command = {
-        bind_rows(
-          forecast_res,
-          ensemble_res %>% mutate(forecaster = "ensemble"),
-          ensemble_mixture_res %>% mutate(forecaster = "ensemble_mix"),
-          # TODO: Maybe later, match with flu_hosp_prod
-          # ensemble_mixture_res_2 %>% mutate(forecaster = "ensemble_mix_2"),
-          # combo_ensemble_mixture_res %>% mutate(forecaster = "combo_ensemble_mix")
+      } else {
+        train_data <-
+          nhsn_latest_data %>%
+          data_substitutions(disease = "covid", last(forecast_generation_dates)) %>%
```

Comment:
Shouldn't this be the particular forecast generation date?
```suggestion
          data_substitutions(disease = "covid", as.Date(forecast_generation_date_int)) %>%
```